docs/aws/audit/lambdamonitoring/rules/dlq_configured.mdx

```diff
@@ -24,6 +24,207 @@ HIPAA, SOC2, PCIDSS, NIST
+
+5. Repeat the process for all the Lambda functions in your AWS account across all regions.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the AWS Lambda functions.
+
+2. Once the AWS CLI is installed and configured, you can list all the Lambda functions in your account by running the following command:
+
+   ```
+   aws lambda list-functions --region your-region
+   ```
+
+   Replace 'your-region' with the region your AWS resources are located in. This command will return a JSON object with all your Lambda functions.
+
+3. To check if a specific Lambda function has a Dead Letter Queue (DLQ) configured, you can use the following command:
+
+   ```
+   aws lambda get-function-configuration --function-name your-function-name --region your-region
+   ```
+
+   Replace 'your-function-name' with the name of the Lambda function you want to check and 'your-region' with the region your AWS resources are located in. This command will return a JSON object with the configuration of the specified Lambda function.
+
+4. In the returned JSON object, look for the 'DeadLetterConfig' attribute. If this attribute is present and has an 'Arn' attribute with a value, then the Lambda function has a DLQ configured. If the 'DeadLetterConfig' attribute is not present or the 'Arn' attribute is empty, then the Lambda function does not have a DLQ configured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services using Python.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access and secret keys.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a client for AWS Lambda service.
+
+```python
+lambda_client = session.client('lambda')
+```
+
+4. Now, you can list all the Lambda functions and check if they have a Dead Letter Queue (DLQ) configured. If the 'DeadLetterConfig' attribute is missing or 'TargetArn' is None, then DLQ is not configured.
+
+```python
+def check_dlq_configured():
+    response = lambda_client.list_functions()
+    for function in response['Functions']:
+        function_name = function['FunctionName']
+        function_details = lambda_client.get_function_configuration(FunctionName=function_name)
+        if 'DeadLetterConfig' not in function_details or not function_details['DeadLetterConfig']['TargetArn']:
+            print(f"Lambda function {function_name} does not have DLQ configured.")
+```
+
+This script will print the names of all Lambda functions that do not have DLQ configured.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>